src/matter/cluster/server/OnOffServer.ts

```diff
@@ -0,0 +1,160 @@
+* AttributeList:
+  * Supported always: onOff, onTime, offWaitTime
+  * Supported if we have a scene cluster: globalSceneControl
+  * Not supported: startUpOnOff
+* AcceptedCommandList:
+  * Supported always: Off, On, Toggle, OnWithTimedOff
+  * Supported if we have a scene cluster: onWithRecallGlobalScene
+  * Supported if we have a Level cluster: OffWithEffect
+* GeneratedCommandList: empty
+* EventList: empty
+* FabricIndex: empty
+ */
+
+/*
+TODO: If the Scenes server cluster is implemented on the same endpoint, the following extension field SHALL
+      be added to the Scene Table:
+      * OnOff
+ */
+
+export const OnOffClusterHandler: () => ClusterServerHandlers<typeof OnOffCluster> = () => ({
+    // TODO In fact for the Cluster server implementation we need "initialization code" or thinks like this.
+    //      I know that it is the wrong place in the "Handlers" object, but I don't know where to put it right now
+    //      because it should be "bound to the cluster instance" and not to the "cluster type".
+    //      Yes could also be build without the subscriptions, but so the code is separated from the other implementation
+    //      parts and if used as library and the "specific device implementation" manipulates onTime or offWaitTime then
+    //      it is all handled automatically and pot. not missed.
+    constructor: ({ attributes: { onOff, onTime, offWaitTime, globalSceneControl } }) => {
+        this.onTimeTimeout: NodeJS.Timeout | null = null;
+        onTime.addMatterListener((value) => {
+            if (this.onTimeTimeout) {
+                clearTimeout(this.onTimeTimeout);
+                this.onTimeTimeout = null;
+            }
+            if (value === 0) {
+                offWaitTime.set(0);
+                onOff.set(false);
+            }
+            if (value > 0 && onOff.get()) {
+                // TODO This is the 100% accurate impelementation right now ...
+                //      maybe could be optimized without a "100ms timeout loop",
+                //      but for now should do exactly what specs say
+                this.onTimeTimeout = setTimeout(() => {
+                    this.onTimeTimeout = null;
+                    onTime.set(value - 1);
+                }, 100);
+            }
+        });
+
+        this.offWaitTimeout: NodeJS.Timeout | null = null;
+        offWaitTime.addMatterListener((value) => {
+            if (this.offWaitTimeout) {
+                clearTimeout(this.offWaitTimeout);
+                this.offWaitTimeout = null;
+            }
+            if (value > 0 && !onOff.get()) {
+                this.offWaitTimeout = setTimeout(() => {
+                    this.offWaitTimeout = null;
+                    offWaitTime.set(value - 1);
+                }, 100);
+            }
+        });
+    },
+    // TODO The trick that we used in Device.ts to access "optionalAttributes" ... how to do it here?
```

Comment:
I (again) need access here to globalSceneControl, but I was not able to get the trick you showed me with "WithOptionalAttributed" to work here

Reply:
I would recommend to implement the scene control in a 2nd pass...

Reply:
For sure - can only be implemented when Scenes are finished, so leaving as Todo for now. Was mainly to tell that we need the inter-cluster-access